gm4_midnight_menaces/data/illusioner_nights/functions/set_pos.mcfunction

```diff
@@ -0,0 +1,18 @@
+#@s = none
+#at world spawn
+#called by event
+
+#spread spawn point around a random player, in a 48 block radius
+execute at @r[gamemode=!spectator] run spreadplayers ~ ~ 0 48 false @e[type=area_effect_cloud,tag=gm4_illusioner_spawn_point]
+
+#set Y position
+execute as @e[type=area_effect_cloud,tag=gm4_illusioner_spawn_point] at @s store result entity @s Pos.[1] double 1 run data get entity @p Pos[1]
+
+execute as @e[type=area_effect_cloud,tag=gm4_illusioner_spawn_point] at @s run tp ~ ~-6 ~
+execute as @e[type=area_effect_cloud,tag=gm4_illusioner_spawn_point] at @s run function illusioner_nights:scan_altitude
+
+#kill locations that are too close to the player
+execute at @a run kill @e[tag=gm4_illusioner_spawn_point,distance=..24]
+execute as @e[type=area_effect_cloud,tag=gm4_illusioner_spawn_point] at @s unless block ~ ~ ~ air unless block ~ ~ ~ cave_air run kill @s
+
+execute as @e[type=area_effect_cloud,tag=gm4_illusioner_spawn_point] at @s run summon illusioner ~ ~ ~ {DeathLootTable:"illusioner_nights:illusioner"}
```

Comment:
You should give them a weapon :P. Bows and tipped arrows maybe?

Reply:
Illusioners naturally carry bows, but I'll check if it can hold and shoot tipped arrows from the offhand

Reply:
Illusioners cannot shoot tipped arrows

Reply:
I am confused then. When I tested this they didn't really attack me (I was in survival). All they did was the blindness spell and the illusion thing, but they did not wield bows or deal any damage to me, they ran around like passive mobs. (Tested this in 1.14)

Reply:
Wait, you're probably right. When adding a DeathLootTable, the entity doesn't spawn with any hand items.

Reply:
It's fixed now